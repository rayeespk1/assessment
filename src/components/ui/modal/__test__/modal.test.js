import React from "react";
import '@testing-library/jest-dom'
import { render, cleanup } from "@testing-library/react";
import Modal from "../modal";

afterEach(cleanup);

test("renders", () => {
    const { asFragment } = render(<Modal show={true}>
        <p>modal content</p>
    </Modal>);
    expect(asFragment()).toMatchSnapshot();
});

test("loading correct data", () => {
    const { getByText } = render(<Modal show={true}>
        <p className="modal-class">modal content</p>
    </Modal>);
    expect(getByText("modal content")).toHaveClass("modal-class");
});