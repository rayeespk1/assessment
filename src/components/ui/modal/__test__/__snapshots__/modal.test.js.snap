// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="Backdrop"
    data-testid="backdropTag"
  />
  <div
    class="Modal"
    style="transform: translateY(0); opacity: 1;"
  >
    <p>
      modal content
    </p>
  </div>
</DocumentFragment>
`;
